### PR TITLE
Allow overriding piece size exponential

### PR DESCRIPTION
**Pull request by evo-33**
I got hit by this when trying to cross seed, figured it might be helpful for others, so here is a pull request.

### CODE REVIEW

**Comment by mueslo** [comment]
What was the actual problem with the piece sizes? Too small? Too big? The current calculation tries to keep the piece size between 32 KiB and 16 MiB, and targets 1024 to 2048 pieces. I thought this would be universally compatible.

File: pythonbits/__init__.py [hunk 1]
```diff
@@ -1,7 +1,7 @@
 # -*- coding: utf-8 -*-
 
 __title__ = "pythonBits"
-__version__ = "3.0b2"
+__version__ = "3.0b3"
```
Comment (mueslo) [inline pythonbits/__init__.py]: please don't bump the version number

File: pythonbits/__main__.py [hunk 2]
```diff
@@ -40,6 +40,8 @@ def n_to_p(x): return "--" + x.replace('_', '-')
                         'help': "Generate screenshots and upload to imgur"},
         'torrentfile': {'short_param': '-t', 'default': False,
                         'help': "Create torrent file"},
+        'piecesize': {'short_param': '-p', 'default': False,
```
Comment (mueslo) [inline pythonbits/__main__.py]: this is unused

File: pythonbits/__init__.py (hunk 1 above)
Comment (evo-33) [reply pythonbits/__init__.py]: Sorted & noted

File: pythonbits/__main__.py (hunk 2 above)
Comment (evo-33) [reply pythonbits/__main__.py]: Ah yes whoops!

**Comment by evo-33** [comment]
The issue with the piece size was with cross-seeding, how you're calculating the piece sizes is perfect. However when cross-seeding I need to have the torrents with different piece sizes. 

>clients identify torrents by their hash, we want to alter it between .torrent files. The easiest way to do that is to use a different piece size 

from /kb.php?id=64

I guess ideally, I'd actually be able to specify the piece size I want, rather than having to deal with exponentials at all, I'll likely get around to that at some point tomorrow. It's late here and I need sleep.

File: pythonbits/__main__.py [hunk 3]
```diff
@@ -67,6 +67,8 @@ def n_to_p(x): return "--" + x.replace('_', '-')
     options_d = {
         'num_screenshots': {'type': int, 'default': 2,
                             'help': "Number of screenshots"},
+        'piece_size_exp': {'type': int, 'default': 0,
```
Comment (mueslo) [inline pythonbits/__main__.py:76]: `None` would be a better choice here

File: pythonbits/mktorrent.py [hunk 4]
```diff
@@ -34,10 +34,11 @@ class MkTorrentException(Exception):
     pass
 
 
-def make_torrent(fname):
+def make_torrent(fname, psize_exp):
```
Comment (mueslo) [inline pythonbits/mktorrent.py]: add default `psize_exp = None`

File: pythonbits/mktorrent.py [hunk 5]
```diff
@@ -34,10 +34,11 @@ class MkTorrentException(Exception):
...
         # todo: multiprocessing
     fsize = get_size(fname)
-    psize_exp = piece_size_exp(fsize)
+    if psize_exp == 0:
```
Comment (mueslo) [inline pythonbits/mktorrent.py]: `if psize_exp is None`

**Comment by mueslo** [comment]
Superseded by f3011625f6ca637e8469367031ba8af339ab4944

**Pull request closed without merging**